src/main/java/Deadline.java

```diff
@@ -0,0 +1,23 @@
+import java.util.Scanner;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public static Task parse(String taskInfo){
+        int i = taskInfo.indexOf(" /by ");
```

Comment:
Maybe it is better to give this variable a more specific name? Since it is not used to iterate through a loop.